uint32 speed
bool direction